File: samples_generated_code/samples_3/athene/baseline_code_c/code_row_102.c
#include <stdio.h>
#include <string.h>

void custom_memcpy(void *dest, const void *src, size_t n) {
    char *d = (char *)dest;
    const char *s = (const char *)src;
    for (size_t i = 0; i < n; i++) {
        d[i] = s[i];
    }
}

int main() {
    char source[] = "Hello, World!";
    char destination[50];
    size_t length = strlen(source) + 1;

    custom_memcpy(destination, source, length);
    printf("%s\n", destination);

    return 0;
}

File: samples_generated_code/samples_3/phi4/baseline_code_java/code_row_102.java
import java.nio.ByteBuffer;

public class MemoryCopyExample {

    public static void main(String[] args) {
        byte[] source = "Hello, World!".getBytes();
        byte[] destination = new byte[20];

        copyBytes(destination, source, 13);

        System.out.println(new String(destination).trim());
    }

    private static void copyBytes(byte[] dest, byte[] src, int numBytes) {
        ByteBuffer destBuffer = ByteBuffer.wrap(dest);
        ByteBuffer srcBuffer = ByteBuffer.wrap(src);

        for (int i = 0; i < numBytes && i < src.length; i++) {
            destBuffer.put(i, srcBuffer.get(i));
        }
    }
}